controllers/mutation_reconcile_looper.go

```diff
@@ -131,7 +142,13 @@ func (m *MutationReconcileLooper) ReconcileMutationObject(ctx context.Context, s
 	}
 	defer vfs.Cleanup(virtualFS)
 
-	if err := utils.ExtractTarToFs(virtualFS, bytes.NewBuffer(resourceData)); err != nil {
+	extractResourceData, err := Ungzip(bytes.NewBuffer(resourceData))
+	if err != nil {
+		// the data is not compressed
+		extractResourceData = resourceData
```

Comment:
This blows my mind. Sometimes it's compressed, sometimes, it isn't.

Reply:
Or rather, it's already decompressed and it is a valid TAR file that needs to be uncompressed.